OpenRA.Mods.Common/AI/States/AirStates.cs

```diff
@@ -31,8 +31,8 @@ protected static int CountAntiAirUnits(IEnumerable<Actor> units)
 			var missileUnitsCount = 0;
 			foreach (var unit in units)
 			{
-				if (unit != null && unit.Info.HasTraitInfo<AttackBaseInfo>() && !unit.Info.HasTraitInfo<AircraftInfo>()
-					&& !unit.IsDisabled())
+				if (unit != null && !unit.Info.HasTraitInfo<AircraftInfo>() &&
+					unit.TraitsImplementing<AttackBase>().Any(ab => !ab.IsTraitPaused && !ab.IsTraitDisabled))
```

Comment:
We can use this opportunity to check against the *correct* armaments, and also reduce the number of queries:
```csharp
if (unit == null || unit.Info.HasTraitInfo<AircraftInfo>())
	continue;

var arms = unit.TraitsImplementing<AttackBase>()
	.Where(ab => !ab.IsTraitPaused && !ab.IsTraitDisabled)
	.SelectMany(ab => ab.Armaments);
```